standard/variables.md

```diff
@@ -901,3 +901,139 @@ variable_reference
+
+For an operator that yields an variable, such as `c ? ref e1 : ref e2`:
+
+- The *ref-safe-to-escape-scope* of the result is the narrowest scope among the *ref-safe-to-escape-scopes* of the operands of the operator.
+- The *safe-to-escape-scope* of the operands must agree, and that is the *safe-to-escape-scope* of the resulting variable.
+
+### §ref-span-safety-method-invocation Method invocation
+
+For a variable `c` resulting from a ref-returning method invocation, `ref e1.M(e2, ...)`, its *ref-safe-to-escape-scope* is the smallest of the following scopes:
+
+- The *calling-method*.
+- The *ref-safe-to-escape-scope* of all `ref` and `out` argument expressions (excluding the receiver).
+- For each `in` parameter of the method, if there is a corresponding expression that is an variable, its *ref-safe-to-escape-scope*, otherwise the nearest enclosing scope
+- The *safe-to-escape-scope* of all argument expressions (including the receiver)
+
+> *Example*: the last bullet is necessary to handle code such as
+>
+> ```csharp
+> var sp = new Span(...)
+> return ref sp[0];
+> ```
+>
+> or
+>
+> ```csharp
+> return ref M(sp, 0);
+> ```
+>
+> *end example*
+
+For a value resulting from a method invocation `e1.M(e2, ...)`, its *safe-to-escape-scope* from the smallest of the following scopes:
+
+- The *calling-method*
+- The *safe-to-escape* of all argument expressions (including the receiver)
+
+A property invocation (either `get` or `set`) it treated as a method invocation of the underlying method by the above rules.
+
+### §ref-span-safety-a-value A value
+
+A value's *ref-safe-to-escape-scope* is the nearest enclosing scope.
+
+> *Note:* This occurs in an invocation such as `M(ref d.Length)` where `d` is of type `dynamic`. It is also consistent with arguments corresponding to `in` parameters.
+
+### §ref-span-safety-stackalloc `stackalloc`
+
+For a value `c` resulting from a `stackalloc` expression its *safe-to-escape-scope* to the current method.
+
+#### §ref-span-safety-constructor-invocations Constructor invocations
+
+A `new` expression that invokes a constructor obeys the same rules as a method invocation (§ref-span-safety-method-invocation) that is considered to return the type being constructed.
+
+In addition, for a value `c` that is the result of a `new` expression, the *safe-to-escape-scope* is no wider than the smallest of the *safe-to-escape-scopes* of all arguments and operands of the object initializer expressions, recursively, if any initializer is present.
+
+### §ref-span-safety-default-expressions `default` expressions
+
+For a `default` expression, the *safe-to-escape-scope* is the *calling-method*.
```

Comment:
```suggestion
The *safe-to-escape-scope* of a `default` expression is the *calling-method*.
```

(This and the previous suggestion just feel like more straightforward statements.)